src/backend/pipeline/fss.c

```diff
@@ -55,19 +45,33 @@ FSSCreateWithMAndH(uint64_t k, TypeCacheEntry *typ, uint64_t m, uint64_t h)
 
 	if (!typ->typbyval)
 	{
+		int i;
+
+		/* For byref types, we use the value as an index into the varlena array */
+		for (i = 0; i < m; i++)
+			fss->monitored_elements[i].value = (Datum ) i;
```

Comment:
Instead of this, we should store their Murmur hash and use that as a faster equality check.